synapse/rest/admin/users.py

```diff
@@ -272,7 +276,9 @@ async def on_PUT(self, request, user_id):
                         target_user.to_string()
                     )
 
-            user = await self.admin_handler.get_user(target_user)
+            user = await self.admin_handler.get_user(
+                target_user
+            )  # type: JsonDict  # type: ignore
```

Comment:
Why ignore this?

Reply:
https://github.com/matrix-org/synapse/blob/aee10768d8a9bf6c3df80859d2c135607aa1bf80/synapse/handlers/admin.py#L60

It is in `get_user` optional, because it can be `NULL` if user does not exist.

https://github.com/matrix-org/synapse/blob/aee10768d8a9bf6c3df80859d2c135607aa1bf80/synapse/rest/admin/users.py#L192-L195
In this case the user always exist an we get a `JsonDict`.

Reply:
I think you want to do `assert user` after this line to let mypy know that it will never be `None`.